### ISSUE TITLE
VSCode does not start on kubuntu 17.10

### ISSUE THREAD

**Issue by dakky** [Microsoft/vscode#43382]
Problem: VSCode does not start on kubuntu 17.10

![image](https://user-images.githubusercontent.com/579584/36061723-a10a4112-0e5e-11e8-8841-93d51dbc2368.png)

- VSCode Version: 1.20.0-1518023506
- OS Version: Kubuntu 17.10

Steps to Reproduce:

1. install deb as stated here: https://code.visualstudio.com/docs/setup/linux
2. launch vscode

Does this issue occur when all extensions are disabled?: Yes

**Comment by duttaditya18** [comment]
Works fine for me. Weird.

**Comment by bpasero** [comment]
Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by duttaditya18** [comment]
@bpasero Cool. Will do that.

**Comment by dakky** [comment]
tried the insider build: 

When starting via menu, nothing happenes. Starting via terminal (debug) seems to be pointless, as the startscript seems the detect the basedir wrong when running in terminal:
```
bash -x code-insiders 
++ id -u
+ '[' 1000 = 0 ']'
+ '[' '!' -L code-insiders ']'
++ dirname code-insiders
+ VSCODE_PATH=./..
+ ELECTRON=./../code-insiders
+ CLI=./../resources/app/out/cli.js
+ ELECTRON_RUN_AS_NODE=1
+ ./../code-insiders ./../resources/app/out/cli.js
/usr/bin/code-insiders: Zeile 35: ./../code-insiders: Datei oder Verzeichnis nicht gefunden
+ exit 127
```
Are there any logs written, when starting code?

**Comment by dakky** [comment]
nope sry.
Running the code directly wihtout the startscript results in this:
```
user@host:~$ /usr/share/code/code /usr/share/code/resources/app/out/cli.js
[main 09:06:41] [ 'TypeError: Cannot convert undefined or null to object',
  '    at Function.keys (<anonymous>)',
  '    at /usr/share/code/resources/app/out/vs/code/electron-main/main.js:37:185',
  '    at Array.forEach (native)',
  '    at Object.t.assign (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:37:151)',
  '    at e.openInBrowserWindow (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:407:734)',
  '    at e.doOpen (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:400:908)',
  '    at e.open (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:398:565)',
  '    at e.startOpenWindow (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:297:415)',
  '    at e.start (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:296:759)',
  '    at e.call (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:295:791)' ]
```

**Comment by bpasero** [comment]
@dakky great now we have a stacktrace that I can point to a location, thanks. Could it be that another instance of Code is running in the background? Maybe try to kill that process and try again?

**Comment by bpasero** [comment]
@joaomoreno @Tyriar fyi for this comment https://github.com/Microsoft/vscode/issues/43382#issuecomment-365165683

**Comment by dakky** [comment]
ah we're getting closer.
**YES** there were some  background/zombie processes:
```
user   15618 15047  0 10:44 pts/1    00:00:00 grep --color=auto code-insiders
user   25726 22309  0 07:17 ?        00:00:10 /usr/share/code-insiders/code-insiders --unity-launch
user   25728 25726  0 07:17 ?        00:00:00 /usr/share/code-insiders/code-insiders --type=zygote --no-sandbox
user   25753 25726  0 07:17 ?        00:00:00 /usr/share/code-insiders/code-insiders --type=gpu-process --no-sandbox --supports-dual-gpus=false --gpu-driver-bug-workarounds=7,23,74 --disable-gl-extensions=GL_KHR_blend_equation_advanced GL_KHR_blend_ t --gpu-vendor-id=0x10de --gpu-device-id=0x1c8d --gpu-driver-vendor=NVIDIA --gpu-driver-version=384.111 --gpu-driver-date --gpu-secondary-vendor-ids=0x8086 --gpu-secondary-device-ids=0x591b --service-request-channel-token=B9623B77E2B6253EDA1CB70FAC7333E6
user   27846 22309  0 07:38 ?        00:00:06 /usr/share/code-insiders/code-insiders --unity-launch
user   27848 27846  0 07:38 ?        00:00:00 /usr/share/code-insiders/code-insiders --type=zygote --no-sandbox
user   27874 27846  0 07:38 ?        00:00:00 /usr/share/code-insiders/code-insiders --type=gpu-process --no-sandbox --supports-dual-gpus=false --gpu-driver-bug-workarounds=7,23,74 --disable-gl-extensions=GL_KHR_blend_equation_advanced GL_KHR_blend_ t --gpu-vendor-id=0x10de --gpu-device-id=0x1c8d --gpu-driver-vendor=NVIDIA --gpu-driver-version=384.111 --gpu-driver-date --gpu-secondary-vendor-ids=0x8086 --gpu-secondary-device-ids=0x591b --service-request-channel-token=BB81AB14E37D1E523F1F41D8C1217D2B
user   27951 22309  0 07:39 ?        00:00:09 /usr/share/code-insiders/code-insiders --unity-launch
user   27953 27951  0 07:39 ?        00:00:00 /usr/share/code-insiders/code-insiders --type=zygote --no-sandbox
user   27977 27951  0 07:39 ?        00:00:00 /usr/share/code-insiders/code-insiders --type=gpu-process --no-sandbox --supports-dual-gpus=false --gpu-driver-bug-workarounds=7,23,74 --disable-gl-extensions=GL_KHR_blend_equation_advanced GL_KHR_blend_ t --gpu-vendor-id=0x10de --gpu-device-id=0x1c8d --gpu-driver-vendor=NVIDIA --gpu-driver-version=384.111 --gpu-driver-date --gpu-secondary-vendor-ids=0x8086 --gpu-secondary-device-ids=0x591b --service-request-channel-token=90C6795E15E16C6DD538464DBA7EF7B5
```

Killing them all and running code via terminal **works**.

When launching code from the menu, ich do **not** get the error window as stated in the report itself, there simply happens nothing.
Checking the processlist shows me again the zombie/background.

TLDR;

The core problem is: running code via kde menu does not show the window, but keeps the process in the background running. Running code again via menu results in the reported error from post 1.

**Comment by bpasero** [comment]
@Tyriar https://github.com/Microsoft/vscode/issues/43382#issuecomment-365210780 might also be interesting for you

**Comment by dakky** [comment]
Can be closed: seems to be a KDE issue:
Starting code via menu works fine **after** rebooting and starting with a clean workingspace/session (restoring the kde workspace is the default)
